compiler/ballerina-lang/src/main/java/io/ballerina/projects/ModuleContext.java

```diff
@@ -409,7 +413,12 @@ static void generateCodeInternal(ModuleContext moduleContext,
         if (bootstrapLangLibName != null) {
             compilerPhaseRunner.performLangLibBirGenPhases(moduleContext.bLangPackage);
         } else {
-            compilerPhaseRunner.performBirGenPhases(moduleContext.bLangPackage);
+            try {
+                compilerPhaseRunner.performBirGenPhases(moduleContext.bLangPackage);
+            } catch (Throwable t) {
+                compilerPhaseRunner.addDiagnosticForUnhandledException(moduleContext.bLangPackage, t);
+                return;
```

Comment:
Don't we need to add this error to the bad-sad log?

Reply:
Added in 1bfe3d6700a474472e3d57098ca04df8126d908d